File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      (String.trim(input) |> String.length) == 0 ->
        "Fine. Be that way!"
      String.at(input, -1) == "?" ->
        cond do
          is_shouting(input) ->
            "Calm down, I know what I'm doing!"
          true ->
            "Sure."
         end
      is_shouting(input) ->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end

  def is_shouting(string) do
    Regex.match?(~r/\p{Lu}/, string) and !Regex.match?(~r/\p{Ll}/, string)
  end
end

File: lua/bob/bob.lua
local bob = {}

function bob.hey(say)
	if say == '' then
		return 'Fine, be that way.'
	end

	local is_yelling  = say == say:upper()
	local is_question = say:sub(#say, #say) == '?'

	if is_question and is_yelling then
		return 'Calm down, I know what I\'m doing!'
	end

	if is_question then
		return 'Sure'
	end

	if is_yelling then
		return 'Whoa, chill out!'
	end

	return 'Whatever'
end

return bob
